DICOM/src/Microsoft.Health.DeID.SharedLib/RedactFunction.cs

```diff
@@ -0,0 +1,147 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using EnsureThat;
+using Microsoft.Health.Dicom.DeID.SharedLib.Models;
+using Microsoft.Health.Dicom.DeID.SharedLib.Settings;
+
+namespace Microsoft.Health.Dicom.DeID.SharedLib
+{
+    public class RedactFunction
+    {
+        private static readonly int AgeThreshold = 89;
+        private static readonly string ReplacementDigit = "0";
+        private static readonly int InitialDigitsCount = 3;
+        private readonly RedactSetting _redactSetting;
+
+        public RedactFunction(RedactSetting redactSetting = null)
+        {
+            _redactSetting = redactSetting ?? new RedactSetting();
+        }
+
+        public string RedactDate(string dateString, string inputDateFormat = null, string outputDateFormat = null, IFormatProvider provider = null)
```

Comment:
🟢 **[Discussion]**
Similar case as that in `DateShiftFunction` class.
`RedactDate` and `RedactDateTime` are almost the same except the DateFormat/DateTimeFormat
Merge them to one function if possible. (unless they have already been used by other developers 😅 )